Plugin priorities

fabianrbz opened the pull request:
### Description

To generate the table, use the priorities defined in `app/_src/.repos/kong-plugins/app/data/priorities/`.
Add a new liquid tag `plugins_priority_table`, that takes `enterprise` or `oss` as an argument and uses the page's release (or latest for Konnect) to fetch the data. 


@lena-larionova I wasn't sure how we wanted to handle the case for `correlation-id`, so I added a separate table for it 🤷. From what I understand `free mode` uses the enterprise value and `oss` well, the oss one.

WDYT?
We could try to replicate the current behaviour too, it's a bit more complex though, but definitely doable.

### Testing instructions

[Preview link](https://deploy-preview-6866--kongdocs.netlify.app/gateway/latest/plugin-development/custom-logic/)

### Checklist 

- [ ] Review label added <!-- (see below) -->
- [ ] For unreleased versions: [conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

    For example, if this change is for an upcoming 3.6 release, surround your change in `{% if_version gte:3.6.x %} <content> {% endif_version %}` tags (or `if_plugin_version` tags for plugins). 


<!-- !!! Only Kong employees can add labels due to a GitHub limitation. If you're an OSS contributor, thank you! The maintainers will label this PR for you !!! -->

<!-- When raising a pull request, indicate what type of review you need with one of the following labels:

    review:copyedit: Request for writer review.
    review:general: Review for general accuracy and presentation. Does the doc work? Does it output correctly?
    review:tech: Request for technical review for a docs platform change.
    review:sme: Request for review from an SME (engineer, PM, etc).

At least one of these labels must be applied to a PR or the build will fail.
-->

lena-larionova commented:
>  I wasn't sure how we wanted to handle the case for correlation-id, so I added a separate table for it 🤷. From what I understand free mode uses the enterprise value and oss well, the oss one.

I wonder if it makes more sense to just remove Correlation ID from the OSS/Free tab, and change the tabs to "OSS package"/"Enterprise package". Because "free mode" is just the Enterprise gateway package without a license, so there's no access to certain plugins. 

So we could do something like:

Tab 1: OSS (or OSS package? Honestly, maybe just OSS)
Note: "If you're running Kong Gateway Enterprise in Free mode, see the Enterprise tab".

Tab 2: Enterprise (same thought as above)


This way, we can add Correlation ID into the main tables and just ignore the "free mode" difference.

fabianrbz commented:
@lena-larionova I've updated it so that:
* we render the plugins and their priorities as they are defined in the json file
* changed the tab to `OSS`

Can you update the [note](https://github.com/Kong/docs.konghq.com/blob/main/app/_src/gateway/plugin-development/custom-logic.md?plain=1#L450-L453) ?

lena-larionova commented:
@fabianrbz made those updates, but oddly enough, the list is missing the saml plugin 🤔 The plugin that kicked all of this off in the first place.

lena-larionova reviewed (approved):
pre-approving, just needs a rebase i think. the plugin priority changes work great though!
